src/lib/wrapAISDK.ts

```diff
@@ -0,0 +1,229 @@
+
+  /**
+   * Scorecard API base URL
+   * Defaults to https://api2.scorecard.io
+   */
+  apiBaseUrl?: string;
+
+  /**
+   * Service name for telemetry
+   * Defaults to "ai-sdk-app"
+   */
+  serviceName?: string;
+
+  /**
+   * Max export batch size for the batch span processor
+   * Defaults to 1 so the traces are exported immediately
+   */
+  maxExportBatchSize?: number;
+}
+
+let tracerProvider: NodeTracerProvider | null = null;
+let tracer: Tracer | null = null;
+const SCORECARD_API_KEY = process.env['SCORECARD_API_KEY'];
+const SCORECARD_PROJECT_ID = process.env['SCORECARD_PROJECT_ID'];
+
+/**
+ * Initialize the OpenTelemetry tracer with Scorecard configuration
+ */
+function initializeTracer(config: ScorecardConfig = {}): Tracer {
+  // Return existing tracer if already initialized
+  if (tracer && tracerProvider) {
+    return tracer;
+  }
+
+  const {
+    projectId = SCORECARD_PROJECT_ID,
+    apiKey = SCORECARD_API_KEY,
+    endpoint = 'https://tracing.scorecard.io/otel/v1/traces',
+    serviceName = 'ai-sdk-app',
+    maxExportBatchSize = 1,
+  } = config;
+
+  // Create resource with service information
+  const resource = defaultResource().merge(
+    resourceFromAttributes({
+      [ATTR_SERVICE_NAME]: serviceName,
+      [ATTR_SERVICE_VERSION]: '1.0.0',
+      'scorecard.project_id': projectId,
+    }),
+  );
+
+  // Create span processors
+  const spanProcessors = [];
+
+  try {
+    // Create OTLP exporter for Scorecard
+    const otlpExporter = new OTLPTraceExporter({
+      url: endpoint,
+      headers:
+        apiKey ?
+          {
+            Authorization: `Bearer ${apiKey}`,
+          }
+        : {},
+    });
+
+    // Add batch span processor with OTLP exporter
+    spanProcessors.push(
+      new BatchSpanProcessor(otlpExporter, {
+        maxExportBatchSize,
+      }),
+    );
+  } catch (error) {
+    // Handle initialization errors
+    console.error('Failed to initialize Scorecard telemetry:', error);
+  }
+
+  // Create tracer provider with span processors
+  tracerProvider = new NodeTracerProvider({
+    resource,
+    spanProcessors,
+  });
+
+  // Register the provider
+  tracerProvider.register();
+
+  // Get tracer instance
+  tracer = tracerProvider.getTracer('ai-sdk-wrapper');
+
+  return tracer;
+}
+
+/**
+ * Wraps the AI SDK module to automatically inject telemetry configuration
+ *
+ * @param aiSDKModule - The AI SDK module (e.g., import('ai'))
+ * @param config - Optional Scorecard configuration
+ * @returns Proxied AI SDK module with automatic telemetry injection
+ *
+ * @example
+ * ```typescript
+ * import ai from 'ai';
+ * import { wrapAISDK } from 'scorecard-ai/lib/wrapAISDK';
+ *
+ * const aiSDK = await wrapAISDK(ai);
+ *
+ * // Now all AI SDK calls will automatically send traces to Scorecard
+ * const { text } = await aiSDK.generateText({
+ *   model: openai('gpt-4o-mini'),
+ *   prompt: 'Hello!',
+ * });
+ * ```
+ */
+export async function wrapAISDK<T extends Record<string, any>>(
```

Comment:
idk if this right, but:

```suggestion
export async function wrapAISDK<T extends Record<string, unknown>>(
```